client.go

```diff
@@ -302,15 +302,25 @@ func (c *Client) CreateDataset(options *CreateDatasetOptions) (datasetState *Dat
 	tickChan := time.NewTicker(tickerWaitingForVolume).C
 
 	for {
-		if s, err := c.GetDatasetState(p.DatasetID); err == nil {
+		var strErrDel string
+		s, err := c.GetDatasetState(p.DatasetID)
+		if err == nil {
 			return s, nil
 		} else if err != errStateNotFound {
-			return nil, err
+			errDel := c.DeleteDataset(p.DatasetID)
```

Comment:
I am not too sure how to handle this correctly. In terms of Kubernetes I am not really caring, as a failed provisioning will be retried endlessly..